icechunk/src/repository.rs

```diff
@@ -73,6 +73,7 @@ pub type RepositoryResult<T> = Result<T, RepositoryError>;
 #[derive(Debug, Serialize, Deserialize)]
 pub struct Repository {
     config: RepositoryConfig,
+    storage_settings: storage::Settings,
```

Comment:
Why is this both in the config and here?

Reply:
Excellent question, i'll make a comment in the code. The settings in config are what the user passed, but most of the time they won't pass anything. `storage_settings` is the "merge" between what users passed and the defaults, so what the repo will use in future calls. We need to retain what the user passed (config) in case they call `save_config`, so I cannot override it with a default.